src/index.tsx

```diff
@@ -306,37 +341,41 @@ export default class SwipeCards extends Component<Props, State> {
 
       currentIndex[this.guid] = 0
       this.setState({
-        cards: [].concat(nextProps.cards),
+        cards: [...nextProps.cards],
         card: nextProps.cards[0],
       })
     }
   }
 
-  _forceLeftSwipe = () => {
+  private forceLeftSwipe = () => {
     const { onCardRemoved, onLeftSwipe } = this.props
     const { card, pan } = this.state
 
-    const choice = Promise.resolve(onLeftSwipe(card))
+    const choice = Promise.resolve(onLeftSwipe!(card, 'button_press'))
 
     const PAN_VALUE = !!onLeftSwipe ? -50 : -500
 
     this.cardAnimation = Animated.timing(pan, {
       toValue: { x: PAN_VALUE, y: 0 },
     }).start((status) => {
       this.state.pan.stopAnimation(() => {
-        choice.then((cancelled) => {
+        choice.then((cancelled: boolean) => {
           if (status.finished && !cancelled) {
-            this.cardAnimation = Animated.timing(pan, {
-              toValue: { x: -500, y: 0 },
-            }).start((status) => {
-              if (status.finished) {
-                this._advanceState()
-                onCardRemoved(currentIndex[this.guid], 'left')
-              } else this._resetState()
-            })
+            if (this.props.smoothTransition) {
+              this.advanceState()
```

Comment:
We weren't respecting the `smoothTransition` prop here, which means that when you pushed the "pass for now" button to swipe left, the animation would flicker that way too. Now, when `smoothTransition` is on it'll be a good animation no matter how the swipe gets triggered